docs/sphinx/manual/building_hpx.rst

```diff
@@ -1357,3 +1357,21 @@ or Unix variants.
 .. include:: ../../generated/cmake_toolchains.rst
 
 .. include:: ../../generated/cmake_variables.rst
+
+
+CMAKE Variable
+==============
+
+This lists the CMAKE variables and how user can enable CMAKE variables during
+make.
+
+- `PLUGIN` : This component is a plugin-able library.
+- `TYPE` : The type can be: EXECUTABLE, LIBRARY or COMPONENT.
+- `INSTALL`: Generates a install rule for the target.
```

Comment:
These are not 'cmake variables' (whatever this means). Using them on the cmake command line will have no effect. The important variables that can be passed in the form of `-DSOME_VARIABLE=<value>` on cmake's command line are listed in the generated file `generated/cmake_variables.rst`. 

What was your rationale to add this text?

Reply:
Aren't these `hpx_setup_target(target)` macro parameters? applicable on https://github.com/STEllAR-GROUP/hpx/issues/1258#issue-42727647

Reply:
@tapaswenipathak I'd say these are already documented [here](https://stellar-group.github.io/hpx/docs/sphinx/branches/master/html/manual/creating_hpx_projects.html#cmake-macros-to-integrate-hpx-into-existing-applications). Would you agree?

Reply:
Sure, can you tell me what should be documented for https://github.com/STEllAR-GROUP/hpx/issues/1258?

Reply:
@tapaswenipathak see my comment on the issue.

Reply:
Reading your comment on the issue I think  we should close this PR @msimberg?